src/ocaml/typing/cmi_format.ml

```diff
@@ -35,7 +35,7 @@ type cmi_infos = {
 }
 
 let input_cmi ic =
-  let (name, sign) = (input_value ic : header) in
+  let (name, sign) = (Ocaml_compression.input_value ic : header) in
```

Comment:
It's how it's done in the compiler: https://github.com/ocaml/ocaml/blob/5.1/file_formats/cmt_format.ml#L108

It's actually a simple call to `Stdlib.input_value` that looks like it will be able to input both compressed (if supported) and uncompressed data.